### ISSUE TITLE
Adding datatype explicitly

### ISSUE THREAD

**Issue by tpluscode** [wikibus/JsonLD.Entities#3]
When a class is serialized `DateTime` values should include `@type` so that they can be interpreted correctly and not as `xsd:string`. *This could be an opt-in feature*.

``` c#
public class Model
{
   public DateTime DateTime { get; set; }
}
```
should produce
``` json
{
   "dateTime": {
      "@value": "2016-03-21T00:00:00",
      "@type": "http://www.w3.org/2001/XMLSchema#dateTime"
   }
}
```

**Comment by asbjornu** [comment]
Isn't this a sensible default? I would expect the same behaviour for all [primitive types](https://www.w3.org/TR/xmlschema-2/#built-in-primitive-datatypes) that can be deduced from the .NET type system.

**Comment by tpluscode** [comment]
Maybe you're right. Maybe except bool and string, which are implicitly typed and unambiguous. Not so with numbers though, because `Number` in JS is used to represent decimals, floats and integers alike.

Would you explicitly serialize the `@type` of every an all primitive type? Or do you think it would be fine to make an exception for `true`/`false` and strings?

**Comment by asbjornu** [comment]
I agree; `string` and `bool` don't need `@type`. As .NET has a richer set of types for numbers than JavaScript, I think differentiating between [`float`](https://www.w3.org/TR/xmlschema-2/#float), [`double`](https://www.w3.org/TR/xmlschema-2/#double), [`decimal`](https://www.w3.org/TR/xmlschema-2/#decimal) and [`integer`](https://www.w3.org/TR/xmlschema-2/#integer) is both valuable and easy to do.
